### PR TITLE
 DEV: Debundle plugin javascript and CSS assets

### PR DESCRIPTION
**Pull request by vinothkannans**
And don't load JavaScript and CSS assets if the plugin is disabled.

### CODE REVIEW

File: spec/requests/list_controller_spec.rb [hunk 1]
```diff
@@ -642,18 +642,27 @@
   end
 
   describe "safe mode" do
+    before do
+      plugin_fixtures = Plugin::Instance.find_all("#{Rails.root}/spec/fixtures/plugins")
+      Discourse.stubs(:plugins).returns(plugin_fixtures)
```
Comment (SamSaffron) [inline spec/requests/list_controller_spec.rb]: this is a lot of magic stubbing... can we do any of this without stubbing?

**Comment by SamSaffron** [comment]
I feel a bit uncomfortable only doing half here, can we add CSS as well?

**Comment by tgxworld** [comment]
@vinothkannans Looks like there are a couple of issues to sort out here. Can you ping me when it is ready for another round of reviews? Thank you :)

**Comment by vinothkannans** [comment]
@tgxworld sure 👍

**Comment by SamSaffron** [comment]
@vinothkannans  any updates here?

**Comment by vinothkannans** [comment]
Not yet. I'm working in missing post uploads and S3 upload stats mostly. I will work on this next week.

**Comment by ZogStriP** [comment]
@vinothkannans any news?

**Comment by tgxworld** [comment]
@ZogStriP @vinothkannans is being blocked on uploads recovery and monitoring at the moment so this might have to wait awhile.

**Comment by vinothkannans** [comment]
> I feel a bit uncomfortable only doing half here, can we add CSS as well?

It's done. @SamSaffron

**Comment by tgxworld** [comment]
@vinothkannans  The specs are broken here.

**Comment by vinothkannans** [comment]
I'm unable to fix this travis build issue. I think it's better to split this PR for plugin JS and CSS files. In that case I may get better way to resolve this problem. I hope it is not an issue.

**Comment by tgxworld** [comment]
@vinothkannans  Sure let do that instead. I can review the PR after our call.

**Comment by vinothkannans** [comment]
@tgxworld I added the JS assets debundling PR in https://github.com/discourse/discourse/pull/7566.

**Pull request closed without merging**